web/assignments.go

```diff
@@ -74,6 +79,55 @@ func (s *AutograderService) deleteCriterion(query *pb.GradingCriterion) error {
 	return s.db.DeleteCriterion(query)
 }
 
+func (s *AutograderService) loadCriteria(ctx context.Context, sc scm.SCM, request *pb.LoadCriteriaRequest) ([]*pb.GradingBenchmark, error) {
+
+	assignment, err := s.db.GetAssignment(&pb.Assignment{ID: request.AssignmentID, CourseID: request.CourseID})
```

Comment:
The method `loadCriteria` is 47 lines long. Consider if lines 84-92 can be made into its own method, but only if there are other similar use cases for getting assignment and course objects. Alternatively, lines 84-103 could also be its own method.

Reply:
You are right, there are a few other use cases where we want an assignment with the corresponding course. Refactored.